examples/simple_regional_with_asm/main.tf

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  cluster_type = "simple-regional-asm2"
+}
+
+provider "google-beta" {
+  version = "~> 3.23.0"
+  region  = var.region
+}
+
+data "google_project" "project" {
+  project_id = var.project_id
+}
+
+module "gke" {
+  source                  = "../../modules/beta-public-cluster/"
+  project_id              = var.project_id
+  name                    = "${local.cluster_type}-cluster${var.cluster_name_suffix}"
+  regional                = true
+  release_channel         = "REGULAR"
+  region                  = var.region
+  network                 = var.network
+  subnetwork              = var.subnetwork
+  ip_range_pods           = var.ip_range_pods
+  ip_range_services       = var.ip_range_services
+  network_policy          = false
+  cluster_resource_labels = { "mesh_id" : "proj-${data.google_project.project.number}" }
+  node_pools = [
+    {
+      name        = "asm-node-pool"
+      autoscaling = false
+      # ASM requires minimum 4 nodes and e2-standard-4
+      # As this is a regional cluster we have node_count * 3 = 6 nodes
+      node_count   = 2
+      machine_type = "e2-standard-4"
+    },
+  ]
+}
+
+module "asm" {
```

Comment:
I'm trying to get this example working in my project and the Terraform command succeeds but it appears that ASM isn't configured properly. Here is the command that I'm running:
```
> terraform apply -auto-approve -var="project_id=<project_id>" -var="region=us-east1" -var="network=" -var="subnetwork=" -var="ip_range_pods=" -var="ip_range_services="
...
Apply complete! Resources: 25 added, 0 changed, 0 destroyed.
```

and here is what I get when I make a request to the `istio-ingressgateway` ingress service:
```
> http http://35.243.147.134
HTTP/1.1 404 Not Found
content-length: 0
date: Mon, 15 Jun 2020 04:56:49 GMT
server: istio-envoy

> http https://35.243.147.134
http: error: ConnectionError: HTTPSConnectionPool(host='35.243.147.134', port=443): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f958b900f98>: Failed to establish a new connection: [Errno 111] Connection refused',)) while doing GET request to URL: https://35.243.147.134/
```

A few additional observations:
* In the `Clusters` Anthos tab, the `simple-regional-asm2-cluster` cluster appears to be healthy
* In the `Service Mesh` Anthos tab, its showing all 11 services, however, theres a banner with a `Failed to load status data for 11 user services.` error
* Clicking through to the `frontend` service, the page shows a `Service data failed to load.` error

Any ideas whats wrong here?

Reply:
@erichaase thanks trying this out and for the updates. You are right that this does not configure an application. I have been smoke testing this by deploying [this sample](https://cloud.google.com/service-mesh/docs/gke-anthos-cli-existing-cluster#installing_a_sample_using_kpt) which seems to be working. Feel free to update this if you run into further issues.